Changes in examples

heisenbuug opened the pull request:
Made change in `mnist_batch_norm.cpp` and `lstm_stock_prediction.cpp`

zoq commented on .travis.yml:
```diff
@@ -79,16 +79,16 @@ script:
-      then
-        continue;
-      fi
+      #if [ "$dir" == "lstm_stock_prediction" ];
+      #then
+      #  continue;
+      #fi
```
Since it's fixed, let's remove the block.

zoq commented on .travis.yml:
```diff
@@ -79,16 +79,16 @@ script:
-      then
-        continue;
-      fi
+      #if [ "$dir" == "mnist_batch_norm" ];
```
Since it's fixed, let's remove the block.

zoq reviewed (commented):
You are awesome, thanks for fixing the examples.

rcurtin commented on mnist_batch_norm/mnist_batch_norm.cpp:
```diff
@@ -186,7 +186,7 @@ int main()
 
   mat testX =
-      tempDataset.submat(0, 1, tempDataset.n_rows - 1, tempDataset.n_cols - 1);
+      tempDataset.submat(1, 0, tempDataset.n_rows - 1, tempDataset.n_cols - 1);
```
Nice catch! :+1:

rcurtin reviewed (approved):
Awesome, looks good!  Thank you for fixing this @heisenbuug!  :100:

I think we may have to open another issue about tuning the batch norm implementation though:

```
Accuracy: train = 63.2407%, valid = 63.6106%
```

63% is... pretty bad for MNIST... :)

heisenbuug commented:
> Awesome, looks good! Thank you for fixing this @heisenbuug! 💯
> 
> I think we may have to open another issue about tuning the batch norm implementation though:
> 
> ```
> Accuracy: train = 63.2407%, valid = 63.6106%
> ```
> 
> 63% is... pretty bad for MNIST... :)

I agree @rcurtin, also actually I wanted to ask why don't we provide a dataset pre-processing library which would contain function like `.info` and `.describe()` in pandas.

zoq reviewed (approved):
Looks good to me, thanks for the fixes.

zoq commented:
> I agree @rcurtin, also actually I wanted to ask why don't we provide a dataset pre-processing library which would contain function like `.info` and `.describe()` in pandas. Having such tools to go through the dataset will make the tuning of models easy as we will get a better understanding of datasets.

I like the idea, I usually use http://arma.sourceforge.net/docs.html#stats_fns and http://arma.sourceforge.net/docs.html#running_stat to get some further information.

rcurtin commented:
Thanks @heisenbuug! :+1:  I saw you worked on the MNIST VAE example too, but it looks like that one's still not working (we can handle that some other time).

heisenbuug commented:
> > I agree @rcurtin, also actually I wanted to ask why don't we provide a dataset pre-processing library which would contain function like `.info` and `.describe()` in pandas. Having such tools to go through the dataset will make the tuning of models easy as we will get a better understanding of datasets.
> 
> I like the idea, I usually use http://arma.sourceforge.net/docs.html#stats_fns and http://arma.sourceforge.net/docs.html#running_stat to get some further information.

Yea, I went through that as well, and I created something like this, actually, I was thinking we could use just using one function to get all the info, I was planning to use these functions in the background only...
![Capture](https://user-images.githubusercontent.com/32970979/99764257-ad38df00-2b22-11eb-9b72-3a16f7220347.PNG)

```
class Data {
private:
  std::string path;
  std::vector<string> headers;
  arma::mat data;

public:
  Data(string file_path)
  {
    path = file_path;
    mlpack::data::Load(path, data, true);
    data = data.submat(0, 1, data.n_rows - 1, data.n_cols - 1);
    getHeaders();
  }

  void info()
  {
    cout << typeid(data).name() << endl;
    std::cout << "Index Range: " << data.n_cols << std::endl;
    std::cout << "Data Columns (total " << data.n_rows << "):" << std::endl;
    for (auto i : headers)
      cout << i << endl;
  }
  void getHeaders()
  {
    std::fstream fin;
    fin.open(path, std::ios::in);
    std::string line, word, temp;
    getline(fin, line);
    stringstream s(line);
    while (getline(s, word, ','))
    {
      headers.push_back(word);
    }
  }

  void infoHelper(int lower, int upper)
  {
    for (auto i : headers)
    {
      cout.width(10);
      cout << i << ", ";
    }

    std::cout << std::endl;
    cout.precision(2);
    cout.setf(ios::fixed);
    cout.width(10);
    data.submat(0, upper, data.n_rows - 1, lower).t().raw_print(cout);
  }

  void head()
  {
    infoHelper(4, 0);
  }

  void tail()
  {
    infoHelper(data.n_cols - 1, data.n_cols - 5);
  }
};
```

This is the code I implemented. Now to call it:
```
int main()
{ 
  Data dataset(../housing.csv");
  dataset.info();
  return 0;
}
```
Are we interested in such functionality? I think data pre-processing and analysis will come in handy for users to take a quick glance at datasets. Shall I open another issue for this @zoq @rcurtin ?

I am thinking of also implementing a `describe`(open to naming suggestions) method which will contain `count mean std min max` and other insightful information. Pandas' methods mostly inspire these and as @zoq suggested we already have many functions available.

rcurtin commented:
@heisenbuug yeah, definitely something like this can be useful, especially in a notebook environment.  Let's continue the discussion in mlpack/mlpack#2722. :+1:
